hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRevokeKycFromAccountHandler.java

```diff
@@ -60,4 +79,32 @@ public TransactionMetadata preHandle(
     public void handle(@NonNull final TransactionMetadata metadata) {
         throw new UnsupportedOperationException("Not implemented");
     }
+
+    /**
+     * Gets the token meta for a given {@link TokenID} and attempts to add a kycRevoke key to the
+     * list of required keys for a given Revoke transaction. Upon failure the status of the {@link
+     * SigTransactionMetadataBuilder} is set to the corresponding {@link ResponseCodeEnum}
+     *
+     * @param tokenId given token id
+     * @param meta given transaction metadata builder
+     * @param tokenStore Provides read-only methods for interacting with data storage for working
+     *     with Tokens.
+     */
+    public void addKycRevokeKey(
+            final TokenID tokenId,
+            final SigTransactionMetadataBuilder meta,
+            @NonNull final ReadableTokenStore tokenStore) {
+        final var tokenMeta = tokenStore.getTokenMeta(tokenId);
+
+        if (!tokenMeta.failed()) {
+            final var kycRevokeKey = tokenMeta.metadata().kycKey();
+            if (kycRevokeKey.isPresent()) {
+                meta.addToReqNonPayerKeys(kycRevokeKey.get());
+            } else {
+                meta.status(ResponseCodeEnum.TOKEN_HAS_NO_KYC_KEY);
```

Comment:
The current logic in SigRequirements [does not set a failing status](https://github.com/hashgraph/hedera-services/blob/d4a6f027e96967a6566224061a460d20f5826e76/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java#L1077) if the token has no kyc key. Rather, we fail in handle.

To be consistent, we need to remove this else block here and maybe just leave something like

tokenMeta.metadata().kycKey().ifPresent(meta::addToReqNonPayerKeys)